### PR TITLE
[RISCV][GISel] Use libcalls for rint, nearbyint, trunc, round, and roundeven intrinsics.

### PR DESCRIPTION
**Pull request by topperc**

### CODE REVIEW

**Comment by tschuett** [comment]
I cannot say much, but the title slightly differs from what you actually legalize, but so what. 4 vs. 5 and some missing fs.

**Comment by topperc** [comment]
> I cannot say much, but the title slightly differs from what you actually legalize, but so what. 4 vs. 5 and some missing fs.

I wrote them in terms of their C and IR names rather than the mishmash we have in G_ node names.

File: llvm/test/CodeGen/RISCV/GlobalISel/legalizer/legalize-fp-libcall.mir [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+    ; CHECK-NEXT: [[COPY1:%[0-9]+]]:_(s64) = COPY $f10_d
+    ; CHECK-NEXT: $f10_d = COPY [[COPY1]](s64)
+    ; CHECK-NEXT: PseudoRET implicit $f10_d
+    %0:_(s64) = COPY $f10_d
+    %1:_(s64) = G_INTRINSIC_ROUND %0
+    $f10_d = COPY %1(s64)
+    PseudoRET implicit $f10_d
+
+...
+---
```
Comment (arsenm) [inline llvm/test/CodeGen/RISCV/GlobalISel/legalizer/legalize-fp-libcall.mir:306]: I think these kind of tests should use IR tests. There's no weird legalizer edge cases being tested, and mir tests are harder to maintain

**Review by arsenm — APPROVED** [review]

File: llvm/test/CodeGen/RISCV/GlobalISel/legalizer/legalize-fp-libcall.mir (hunk 1 above)
Comment (topperc) [reply llvm/test/CodeGen/RISCV/GlobalISel/legalizer/legalize-fp-libcall.mir:306]: Agreed. I'll commit this as is and then try to replace with an IR test as a followup.

**Pull request merged**